File: Book Studio/MatrixBook.Server/Models/DashboardModel.cs
namespace MatrixBook.Server.Models;

public class DashboardModel
{
    public int BookCount { get; set; }
    public int FinishedBookCount { get; set; }
    public int InProgressBookCount { get; set; }
    public long WordCount { get; set; }
    public long FinishedWordCount { get; set; }
    public long UnfinishedWordCount { get; set; }
    public int AuthorCount { get; set; }
    public List<BookInfo> Books { get; set; } = [];
}

File: Book Studio/matrixbook.client/src/app/models/dashboard-model.ts
export interface BookInfo {
    title?: string;
    author?: string;
    status?: string;
    textCount?: number;
    dateCreated?: Date;
    dateUpdated?: Date;
}
export interface DashboardModel {
    bookCount?: number;
    finishedBookCount?: number;
    inProgressBookCount?: number;
    wordCount?: number;
    finishedWordCount?: number;
    unfinishedWordCount?: number;
    authorCount?: number;
    books?: BookInfo[];
}